Packs/SalesforceITAdmin/Integrations/Salesforce_IT_Admin/Salesforce_IT_Admin.py

```diff
@@ -0,0 +1,1164 @@
+
+        custom_mapping = None
+        if args.get('customMapping'):
+            custom_mapping = json.loads(args.get('customMapping'))
+        elif CUSTOM_MAPPING_CREATE:
+            custom_mapping = json.loads(CUSTOM_MAPPING_CREATE)
+
+        extension_schema = scim.get(SCIM_EXTENSION_SCHEMA)
+        if custom_mapping and extension_schema:
+            for key, value in custom_mapping.items():
+                # key is the attribute name in input scim. value is the attribute name of app profile
+                user_extension_data = extension_schema.get(key)
+                if user_extension_data:
+                    salesforce_user[value] = user_extension_data
+        return salesforce_user
+
+    # Builds a new user salesforce profile dict with pre-defined keys and custom mapping (for user)
+
+    def build_salesforce_profile_update_user(self, args, scim):
+        parsed_scim_data = map_scim(scim)
+
+        salesforce_user = {
+            "Username": parsed_scim_data.get('userName'),
+            "Email": parsed_scim_data.get('email'),
+            "FirstName": parsed_scim_data.get('first_name'),
+            "LastName": parsed_scim_data.get('last_name'),
+            "Name": parsed_scim_data.get('Name'),
+            "CommunityNickname": parsed_scim_data.get('nick_name'),
+            "IsActive": parsed_scim_data.get('active'),
+            "Address": parsed_scim_data.get('address'),
+            "Street": parsed_scim_data.get('street'),
+            "City": parsed_scim_data.get('city'),
+            "Country": parsed_scim_data.get('country'),
+            "State": parsed_scim_data.get('state'),
+            "PostalCode": parsed_scim_data.get('zip'),
+            "MobilePhone": parsed_scim_data.get('mobile_phone'),
+            "Phone": parsed_scim_data.get('phone'),
+            "TimeZoneSidKey": parsed_scim_data.get('timezone'),
+            "LocaleSidKey": parsed_scim_data.get('locale'),
+            "LanguageLocaleKey": parsed_scim_data.get('locale'),
+            "Title": parsed_scim_data.get('title'),
+            "FullPhotoUrl": parsed_scim_data.get('full_photo_url'),
+            "SmallPhotoUrl": parsed_scim_data.get('small_photo_url'),
+        }
+
+        custom_mapping = None
+        if args.get('customMapping'):
+            custom_mapping = json.loads(args.get('customMapping'))
+        elif CUSTOM_MAPPING_UPDATE:
+            custom_mapping = json.loads(CUSTOM_MAPPING_UPDATE)
+
+        extension_schema = scim.get(SCIM_EXTENSION_SCHEMA)
+        if custom_mapping and extension_schema:
+            for key, value in custom_mapping.items():
+                # key is the attribute name in input scim. value is the attribute name of app profile
+                user_extension_data = extension_schema.get(key)
+                if user_extension_data:
+                    salesforce_user[value] = user_extension_data
+
+        return salesforce_user
+
+    def build_salesforce_profile_enable_user(self, args, scim):
```

Comment:
can we unify this function and the disable_user one?

Reply:
Done the changes